test/experimental/features/kreference_group/channel_to_channel.go

```diff
@@ -0,0 +1,128 @@
+/*
+Copyright 2021 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kreference_group
+
+import (
+	"context"
+
+	cetest "github.com/cloudevents/sdk-go/v2/test"
+	"github.com/stretchr/testify/require"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	messagingv1 "knative.dev/eventing/pkg/apis/messaging/v1"
+	eventingclient "knative.dev/eventing/pkg/client/injection/client"
+	"knative.dev/eventing/test/rekt/resources/channel"
+	"knative.dev/eventing/test/rekt/resources/subscription"
+	duckv1 "knative.dev/pkg/apis/duck/v1"
+	"knative.dev/reconciler-test/pkg/environment"
+	"knative.dev/reconciler-test/pkg/eventshub"
+	"knative.dev/reconciler-test/pkg/eventshub/assert"
+	"knative.dev/reconciler-test/pkg/feature"
+)
+
+// ChannelToChannel tests a scenario where the flow is source -> A -> B -> sink
+// and A -> B subscription uses the KReference.Group field.
+func ChannelToChannel() *feature.Feature {
+	f := feature.NewFeature()
+
+	channelGVK := channel.GVK()
+	channelGVR := channel.GVR()
+	channelGroup := channelGVK.GroupKind().Group
+	channelAPIVersion, imcKind := channelGVK.ToAPIVersionAndKind()
+
+	channelAName := feature.MakeRandomK8sName("channel-a")
+	channelBName := feature.MakeRandomK8sName("channel-b")
+	subAToBName := feature.MakeRandomK8sName("sub-a-b")
+	subBToSinkName := feature.MakeRandomK8sName("sub-b-sink")
+	sinkName := feature.MakeRandomK8sName("sink")
+	sourceName := feature.MakeRandomK8sName("source")
+
+	ev := cetest.FullEvent()
+
+	f.Setup("install sink", eventshub.Install(
+		sinkName,
+		eventshub.StartReceiver,
+	))
+
+	f.Setup("Install channel A", channel.Install(channelAName))
+	f.Setup("Install channel B", channel.Install(channelBName))
+	f.Setup("channel A is ready", channel.IsReady(channelAName))
+	f.Setup("channel B is ready", channel.IsReady(channelBName))
+
+	f.Setup("Install channel A -> channel B subscription", func(ctx context.Context, t feature.T) {
+		namespace := environment.FromContext(ctx).Namespace()
+		_, err := eventingclient.Get(ctx).MessagingV1().Subscriptions(namespace).Create(ctx,
```

Comment:
Rather than do this, you can use the subscription.Install resource package?

Reply:
TBH I think this code is cleaner. Also I was worried to use it, because in order to do so, I need to modify the template to allow for the new `group` field. I would rather keep it this way and revisit in future.